wagtail/admin/views/reports/page_types_usage.py

```diff
@@ -0,0 +1,175 @@
+        null_label=_("All"),
+        null_value=None,
+    )
+    site = SiteFilter(
+        label=_("Site"),
+        choices=_get_site_choices,
+        empty_label=None,
+        null_label=_("All"),
+        null_value=None,
+    )
+
+    class Meta:
+        model = ContentType
+        fields = ["page_locale", "site"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.sites = {
+            site.pk: site for site in Site.objects.all().prefetch_related("root_page")
+        }
+        self.sites_filter_enabled = True
+        if len(self.sites) == 1:
+            # If there is only one site, we don't need to show the site filter
+            self.sites_filter_enabled = False
+            del self.filters["site"]
+
+
+class PageTypesUsageReportView(ReportView):
+    template_name = "wagtailadmin/reports/page_types_usage.html"
+    title = _("Page types usage")
+    header_icon = "doc-empty-inverse"
+    filterset_class = PageTypesUsageReportFilterSet
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.user_model = get_user_model()
+        self.page_models = [model.__name__.lower() for model in get_page_models()]
+        self.i18n_enabled = getattr(settings, "WAGTAIL_I18N_ENABLED", False)
+
+    def user_id_to_python(self, user_id):
+        return self.user_model._meta.pk.to_python(user_id)
+
+    def add_last_edited_name_to_page_type(self, username_mapping, page_type):
+        if page_type.last_edited_page_owner_id:
+            # Giving the last_published_by annotation an explicit output_field type
+            # causes an issue with prefetch_workflow_states when the field is a
+            # ConvertedValueField. If the last user to publish the page has been
+            # deleted, we will render their user id in the template, so we call
+            # to_python on the value so that what's rendered matches the developer's
+            # expectation in the case of complex primary key types (e.g. UUIDField).
+            try:
+                user_id_value = self.user_id_to_python(
+                    page_type.last_edited_page_owner_id
+                )
+            except ValidationError:
+                user_id_value = page_type.last_edited_page_owner_id
+
+            last_edited_page_owner = username_mapping.get(
+                user_id_value, get_deleted_user_display_name(user_id=user_id_value)
+            )
+            page_type.last_edited_page_owner = last_edited_page_owner
+
+    def add_last_edited_page_to_page_type(self, pages_mapping, page_type):
+        if page_type.last_edited_page_id:
+            page_type.last_edited_page = pages_mapping.get(
+                page_type.last_edited_page_id
+            )
+
+    def decorate_paginated_queryset(self, page_types):
+        page_ids = set(page_types.values_list("last_edited_page_id", flat=True))
+        pages = Page.objects.filter(pk__in=page_ids).select_related("owner")
```

Comment:
This should do the trick:

```suggestion
        pages_by_id = Page.objects.select_related("owner").in_bulk(obj.last_edited_page_id for obj in page_types if obj.last_edited_page_id)
```

Reply:
Neat, I'll try that

Reply:
Done!